testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -5820,3 +5819,313 @@ fn atlas_stress_integration_test() {
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    btc_regtest_controller.bootstrap_chain(200);
+
+    let mut run_loop = neon::RunLoop::new(conf);
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+    let burnchain_height = run_loop.get_burnchain_height_arc();
+
+    let channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // Wait until block 210, so we now that the burnchain is ready.
+    wait_for_runloop(&blocks_processed);
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        210,
+    );
+
+    // Publish the contract so we can use it.
+    submit_tx(
+        &http_origin,
+        &make_contract_publish(
+            &creator_sk,
+            0,
+            1100000,
+            "increment-contract",
+            &giant_contract,
+        ),
+    );
+
+    // Wait to make sure the contract is published.
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        215,
+    );
+
+    // Check that we have defined the contract.
+    let increment_contract_defines = select_transactions_where(
+        &test_observer::get_blocks(),
+        |transaction| match &transaction.payload {
+            TransactionPayload::SmartContract(contract) => {
+                contract.name == ContractName::try_from("increment-contract").unwrap()
+            }
+            _ => false,
+        },
+    );
+    assert_eq!(increment_contract_defines.len(), 1);
+
+    // Alice calls the contract and should succeed, because we have not lowered the block limit
+    // yet.
+    submit_tx(
+        &http_origin,
+        &make_contract_call(
+            &alice_sk,
+            0,
+            1000,
+            &creator_addr.into(),
+            "increment-contract",
+            "increment-many",
+            &[Value::UInt(0 as u128)],
+        ),
+    );
+
+    // Wait for the contract call to process.
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        220,
+    );
+
+    // Check that we have processed the contract successfully, by checking that the contract call
+    // is in the block record.
+    let increment_calls_alice = select_transactions_where(
+        &test_observer::get_blocks(),
+        |transaction| match &transaction.payload {
+            TransactionPayload::ContractCall(contract) => {
+                contract.contract_name == ContractName::try_from("increment-contract").unwrap()
+            }
+            _ => false,
+        },
+    );
+    assert_eq!(increment_calls_alice.len(), 1);
+
+    // Clear the observer so we can look for Bob's transaction.
+    test_observer::clear();
+
+    // The cost contract was switched at height 220. So, now we expect Bob's call to fail.
```

Comment:
Can you test the edge case here?  Does the new block limit come into effect on block 220, or block 221?

Reply:
See my test in `epoch_205::test_dynamic_db_method_costs` -- it does an exact check on the cut over. It does "overrun" a bit, but it explicitly checks the receipts  make sure that the transition is done at the right block.

Reply:
Now stopping on exact block and asserting this

Reply:
Similar to the other comment threads -- the tighter bounds on the burnchain height addresses some of this issue, so if there's still more we want out of this integration test, we should discuss on the PR itself.